x-pack/plugins/observability_solution/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -606,42 +613,101 @@ export const tasks: TelemetryTask[] = [
+              body: {
+                size: 0,
+                track_total_hits: false,
+                timeout,
+                query: {
+                  bool: {
+                    filter: [
+                      {
+                        term: {
+                          [AGENT_NAME]: agentName,
+                        },
                       },
+                      range1d,
+                    ],
+                  },
+                },
+                aggs: {
+                  services: {
+                    cardinality: {
+                      field: SERVICE_NAME,
                     },
-                    range1d,
-                  ],
+                  },
                 },
               },
-              aggs: {
-                services: {
-                  cardinality: {
-                    field: SERVICE_NAME,
+            });
+
+            return {
+              ...data,
+              [agentName]: response.aggregations?.services.value || 0,
+            };
+          });
+        },
+        Promise.resolve({} as Record<AgentName, number>)
+      );
+
+      const initOtelAgents = OPEN_TELEMETRY_AGENT_NAMES.reduce((acc, agent) => {
+        acc[agent] = 0;
+        return acc;
+      }, {} as Record<OpenTelemetryAgentName, number>);
+
+      const servicesPerOtelAgents = await OPEN_TELEMETRY_BASE_AGENT_NAMES.reduce(
+        (prevJob, agentName) => {
+          return prevJob.then(async (data) => {
+            const response = await telemetryClient.search({
+              index: [indices.error, indices.span, indices.metric, indices.transaction],
+              body: {
+                size: 0,
+                track_total_hits: false,
+                timeout,
+                query: {
+                  bool: {
+                    filter: [{ prefix: { [AGENT_NAME]: agentName } }, range1d],
+                  },
+                },
+                aggs: {
+                  agent_name: {
+                    terms: {
+                      field: AGENT_NAME,
+                      size: 1000,
+                    },
+                    aggs: {
+                      services: {
+                        cardinality: {
+                          field: SERVICE_NAME,
+                        },
+                      },
+                    },
                   },
                 },
               },
-            },
-          });
+            });
 
-          return {
-            ...data,
-            [agentName]: response.aggregations?.services.value || 0,
-          };
-        });
-      }, Promise.resolve({} as Record<AgentName, number>));
+            const aggregatedServicesPerAgents = response.aggregations?.agent_name.buckets.reduce(
+              (acc, agent) => {
+                return {
+                  ...acc,
+                  [agent.key as OpenTelemetryAgentName]: agent.services.value || 0,
+                };
+              },
+              initOtelAgents
+            );
```

Comment:
That's the way I did this initially but switched over to spread operator syntax because I saw the other examples followed this pattern (I assumed this was recommended). 

Thank you for the tip, I didn't know about GC pressure. I'll update it and others!